src/main/java/com/example/bookstore/controller/OrderController.java

```diff
@@ -0,0 +1,80 @@
+package com.example.bookstore.controller;
+
+import com.example.bookstore.dto.CreateOrderRequestDto;
+import com.example.bookstore.dto.OrderItemResponseDto;
+import com.example.bookstore.dto.OrderResponseDto;
+import com.example.bookstore.dto.UpdateOrderRequestDto;
+import com.example.bookstore.model.User;
+import com.example.bookstore.service.OrderService;
+import com.example.bookstore.service.UserService;
+import io.swagger.v3.oas.annotations.Operation;
+import io.swagger.v3.oas.annotations.tags.Tag;
+import jakarta.validation.Valid;
+import java.util.List;
+import java.util.Set;
+import lombok.RequiredArgsConstructor;
+import org.springframework.http.HttpStatus;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PatchMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.ResponseStatus;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequestMapping("/api/orders")
+@RequiredArgsConstructor
+@Tag(name = "Order management", description = "Endpoints for managing orders")
+public class OrderController {
+    private final OrderService orderService;
+    private final UserService userService;
+
+    @PreAuthorize("hasRole('ROLE_USER')")
+    @PostMapping
+    @ResponseStatus(HttpStatus.CREATED)
+    @Operation(summary = "Create a new order")
+    OrderResponseDto createOrder(Authentication auth,
+                                 @RequestBody @Valid CreateOrderRequestDto requestDto) {
+        return orderService.createOrder(getUser(auth), requestDto);
```

Comment:
Could this realization be considered stateless? You're getting user id from the authentication object. Let's get it from URL parameter. You can use the next URL pattern: `.../users/{userId}/orders...` Please fix the same issue in all places